Update ZHA section about Zigbee OTA (#28895)
home-assistant/home-assistant.github.io: source/_integrations/zha.markdown

Additional devices in the [Known working Zigbee radio modules](#known-working-zigbee-radio-modules) list may be discoverable, however, only devices that have been confirmed discoverable are listed above.

### OTA updates of Zigbee device firmware

> The ZHA integration have the ability to perform OTA (Over-The-Air) firmware updates of paired/joined Zigbee devices and this feature is enabled by default. As this uses standard [Update](/integrations/update/) entities in Home Assistant, Home Assistant users will get a UI notification if and when an OTA firmware update is available for a specific device, with an option to initiate the update or ignore that specific update for that device.

Reviewer: ```suggestion
The ZHA integration can perform Over-The-Air (OTA) firmware updates of Zigbee devices. This feature is enabled by default. As this uses standard [Update](/integrations/update/) entities in Home Assistant, Home Assistant users will get a UI notification when an OTA firmware update is available for a specific device, with an option to initiate the update or ignore that specific update for that device.
```

> The ZHA integration have the ability to perform OTA (Over-The-Air) firmware updates of paired/joined Zigbee devices and this feature is enabled by default. As this uses standard [Update](/integrations/update/) entities in Home Assistant, Home Assistant users will get a UI notification if and when an OTA firmware update is available for a specific device, with an option to initiate the update or ignore that specific update for that device.

Reviewer 2: ```suggestion
The ZHA integration has the ability to perform OTA (over-the-air) firmware updates of Zigbee devices. This feature is enabled by default. As it uses standard [Update](/integrations/update/) entities in Home Assistant, users will get a UI notification if and when an OTA firmware update is available for a specific device, with an option to initiate the update or ignore that specific update for the device.
```

> Prerequisites for OTA updates to become available for a specific device is that it both support the standard Zigbee OTAU (Over-The-Air Update) feature and firmware images with standard Zigbee OTA format for that brand and device model are provided to the public by the manufacturer. For this reason ZHA currently only includes OTA providers from a few manufacturers that provide official online sources for Zigbee device OTA firmware images for their brands and models. Today that is; IKEA, INOVELLI, LEDVANCE (OSRAM), SALUS (Computime), Sonoff (ITead), and 3reality (Third Reality).

Reviewer 2: ```suggestion
To see OTA updates for a device, it's required that it both supports OTA updates and that firmware images for the device are publicly provided by the manufacturer. For this reason, ZHA currently only includes OTA providers for a few manufacturers that provide these updates publicly. This includes IKEA, Inovelli, Ledvacnce/OSRAM, SALUS/Computime, Sonoff/iTead, and Third Reality.
```

<div class="note warning">

Before updating a device, you should search for any disadvantages or if you even need to install an available update. Some firmware updates can break features you might use (e.g. group binding for IKEA devices). Some updates might also require changes to ZHA. In rare cases, you can even brick devices by installing a firmware update.

> </div>

Author: This warning already does that

```
<div class="note warning">

Before updating a device, you should search for any disadvantages or if you even need to install an available update. Some firmware updates can break features you might use (e.g. group binding for IKEA devices). Some updates might also require changes to ZHA. In rare cases, you can even brick devices by installing a firmware update.

</div>
```

> When it comes to Zigbee device OTA firmware updates it is always recommended to check if an update will bring you any advantages before updating. Most Zigbee device OTA firmware updates are normally NOT specifically made for this ZHA integration, but instead usually only tested to work with the manufacturer's own Zigbee gateway/hub/bridge from the same brand. In some cases, the firmware update can cause unwanted issues, including no longer accessible functions until ZHA Device Handlers (quirks) are added/updated, or at worse even bricking a device causing it to stop working. So the general advice is to try to apply a "*if it ain't broke then don't fix it*" philosophy when it comes to Zigbee device firmware updates.

Reviewer 2: ```suggestion
Before updating a device, you should search for any disadvantages or if you even need to install an available update. Some firmware updates can break features you might use (e.g. group binding for IKEA devices). Some updates might also require changes to ZHA. In rare cases, you can even brick devices by installing a firmware update.
```

  type: string
{% endconfiguration %}

### OTA updates of Zigbee device firmware

> The ZHA integration has the ability to perform firmware updates of Zigbee devices if they both support the standard Zigbee OTAU (Over-The-Air Update) feature and firmware images with standard Zigbee OTA format are provided by the manufacturer. Firmware images for some device brands can be downloaded automatically while for other brands of devices have to be manually downloaded or provided by other methods.

Reviewer: We don't want to instruct people to download any OTA files. This is an advanced feature and can be discovered elsewhere, not in the main ZHA documentation.

Author: OK removed

> OTA providers from a few manufacturer's official online sources that enable automatic download and installation of firmware update images are currently only available for Zigbee device brands from; IKEA, INOVELLI, LEDVANCE (OSRAM), SALUS (Computime), Sonoff (ITead), and 3reality (Third Reality). Automatic OTA firmware updating is set to disabled in the configuration by default, and if you set it to enabled for a brand then ZHA will currently automatically update firmware without notifications on any compatible device as soon as it finds a compatible image available on local storage.

Reviewer: This isn't right: ZHA doesn't auto-update devices anymore. We download OTA images and you have to start the update by clicking a button.

Author: Yes changed that now, (this PR was originally submitted in beginning of September 2023 and much/most has changed since).

Reviewer 3: This still isn't really correct. As far as I know, there is no automatic firmware update any more, so there is also nothing that could be enabled in the configuration.

> OTA providers from a few manufacturer's official online sources that enable automatic download and installation of firmware update images are currently only available for Zigbee device brands from; IKEA, INOVELLI, LEDVANCE (OSRAM), SALUS (Computime), Sonoff (ITead), and 3reality (Third Reality). Automatic OTA firmware updating is set to disabled in the configuration by default, and if you set it to enabled for a brand then ZHA will currently automatically update firmware without notifications on any compatible device as soon as it finds a compatible image available on local storage.

Reviewer 3: Ikea is disabled by default, see https://github.com/zigpy/zigpy/blob/dev/zigpy/config/defaults.py#L26 and https://github.com/zigpy/zigpy/commit/9a79788632b29345f36af5dd11b89d85d9900e85 (from https://github.com/zigpy/zigpy/pull/1321)
I assume that's also why Ikea was not mentioned in the release notes (https://www.home-assistant.io/blog/2024/02/07/release-20242/#updating-your-zigbee-devices).

Author: Yes, IKEA is currently disabled by default, but I do not think that is something that should be in the ZHA documentation as I believe it is only temporarly (or until further notice via the changelog), or?

Reviewer 3: I don't know. Would need information from @dmulcahey since he disabled it.

Author: @puddly Should it specifically be mentioned in ZHA documentation that OTA for IKEA is currently disabled by default?

Author: Please re-review latest commit with rewrite https://github.com/home-assistant/home-assistant.io/pull/28895/commits/50f9fe399bb5f94701f56a6bb857f6e23fde3603

> OTA firmware updates can be manually downloaded and initiated. Updates from a local OTA update directory can be used as an option for offline firmware updates if the user provides the correct Zigbee OTA formatted firmware files themselves.

Reviewer: This isn't something we want in the documentation. File-based OTA is dangerous and not recommended.

Author: OK removed

> Support for additional online OTA providers for other manufacturers can either be added as OTA firmware providers by developers as a new OTA provider with download code in the zigpy library, if the device manufacturer publishes their firmware images publicly via OTA firmware provider source URL, or by anyone without extra code as a generic zigpy OTA provider via "remote providers" if create/generate a compatible index file in JSON format, which can be used to provide OTA updates over a local network or from the internet (this allows for manufacturers to distribute test feed URLs to customers without deploying them globally).

Reviewer: This is more developer-focused and doesn't belong in the ZHA docs.

Author: OK removed

- https://github.com/zigpy/zigpy/wiki/OTA-Device-Firmware-Updates
- https://github.com/zigpy/zigpy/wiki/OTA-Information-for-Manufacturers
- https://github.com/zigpy/zigpy/blob/dev/README.md#zigbee-device-ota-updates
- https://github.com/zigpy/zigpy/tree/dev/zigpy/ota

> - https://github.com/zigpy/zigpy-cli/blob/dev/README.md#ota

Reviewer: Nor are these URLs.

Author: OK removed

<div class="note warning">

Always check if a firmware update will actually bring you advantages. Most firmware updates are normally NOT specifically made for this ZHA integration, but instead usually only tested to work with the manufacturer's own Zigbee gateway/hub/bridge from the same brand. In some cases, the firmware update can cause unwanted issues, including no longer accessible functions until ZHA Device Handlers (quirks) are added/updated, or at worse even bricking a device causing it to stop working.

</div>

> To enable OTA firmware updates for the ZHA integration you need to add the following configuration to your `configuration.yaml` and restart Home Assistant.

Reviewer: ZHA OTA doesn't have any YAML configuration anymore unless you want to disable it, this section (and the code snippet below) can be removed to make the docs shorter.

Reviewer 3: You could also enable Ikea updates, which are disabled by default, in YAML configuration. Or is this something you don't want to have in the documentation?

Author: Changed so that it instead describe how to disable OTA providers

  type: string
{% endconfiguration %}

### OTA updates of Zigbee device firmware

> The ZHA integration have the ability to perform firmware updates of some specific Zigbee devices if they both support the standard Zigbee OTAU (Over-The-Air Update) feature and firmware images with standard Zigbee OTA format that brand and device are provided to the public by the manufacturer. You will get notifications if and when a device has firmware update available and to start an upgrade you have to initiate it by click a button.

Reviewer 4: Maybe worth to mention that updates can be skipped as well.

Author: Please re-review latest commit with rewrite https://github.com/home-assistant/home-assistant.io/pull/28895/commits/50f9fe399bb5f94701f56a6bb857f6e23fde3603

> OTA providers from a few manufacturer's official online sources that allow firmware update images are currently only available for Zigbee device brands from; IKEA, INOVELLI, LEDVANCE (OSRAM), SALUS (Computime), Sonoff (ITead), and 3reality (Third Reality). Automatic OTA firmware updating is set to disabled in the configuration by default, and if you set it to enabled for a brand then ZHA will currently automatically update firmware without notifications on any compatible device as soon as it finds a compatible image available on local storage.

Reviewer 4: Unclear for me at least: how to enable automatic updates? Same provider tags described below to disable provider updates? Or something else? 
If they are the same options, it's a little unexpected. Because the config is a tri state then: (true, undefined, false) ?!

Reviewer: This isn't correct, there is no way to enable automatic updates. ZHA uses standard Home Assistant update entities and it's no different from any other integration in this regard.

Author: Please re-review latest commit with rewrite https://github.com/home-assistant/home-assistant.io/pull/28895/commits/50f9fe399bb5f94701f56a6bb857f6e23fde3603

  type: string
{% endconfiguration %}

### Advanced OTA configuration

> The default configuration for  OTA (Over-The-Air) firmware updates is today carefully chosen by ZHA developers, so normal users should not need to change any configurations manually. As such almost all of the configuration options listed in the zigpy section in the Home Assistant `configuration.yaml` file are just meant for development or advanced users. 

Reviewer 2: ```suggestion
The default configuration for OTA firmware updates is chosen by ZHA developers, so normal users should not need to change any configuration. Most of the config options listed in the zigpy section are just meant for development or advanced users.
```

> Hence most advanced configuration options will remain in the [zigpy project's developers documentation](https://github.com/zigpy/zigpy). The exception is if want to disable OTA firmware updates for a specific manufacturer's provider, as then you can add the following configuration to your `configuration.yaml` and restart Home Assistant.

Reviewer 2: ```suggestion
Further advanced configuration options are only provided in the [zigpy project's developers documentation](https://github.com/zigpy/zigpy).
However, if you want to disable OTA updates for a specific manufacturer, you can add the following lines to your `configuration.yaml` and restart Home Assistant.
```

  type: string
{% endconfiguration %}

### Advanced OTA configuration

The default configuration for OTA firmware updates is chosen by ZHA developers, so normal users should not need to change any configuration. Most of the config options listed in the zigpy section are just meant for development or advanced users.

Further advanced configuration options are only provided in the [zigpy project's developers documentation](https://github.com/zigpy/zigpy).

However, if you want to disable OTA updates for a specific manufacturer, you can add the following lines to your `configuration.yaml` and restart Home Assistant.

```yaml
zha:
  zigpy_config:
    ota:
      ikea_provider: false                       # Disable OTA update downloads for Trådfri devices
      inovelli_provider: false                   # Disable OTA update downloads for INOVELLI devices
      ledvance_provider: false                   # Disable OTA update downloads for LEDVANCE/OSRAM devices
      salus_provider: false                      # Disable OTA update downloads for SALUS/Computime devices
      sonoff_provider: false                     # Disable OTA update downloads for Sonoff (ITead) devices

>       thirdreality_provider: false               # Disable OTA update downloads for 3REALITY devices

Author: This is documentation that reflect the core code,

  zigpy_config:
    ota:
      ikea_provider: true                        # Auto update Trådfri devices
      inovelli_provider: true                    # Auto update INOVELLI devices
      ledvance_provider: true                    # Auto update LEDVANCE/OSRAM devices
      salus_provider: true                       # Auto update SALUS/Computime devices
      sonoff_provider: true                      # Auto update Sonoff (ITead) devices
      thirdreality_provider: true                # Auto update 3REALITY devices
      #otau_directory: /path/to/your/ota/folder  # Utilize .ota files to update everything else
```

> To use generic OTA providers for zigpy in the ZHA integration you need to provide URLs to your generated JSON file(s) using the following configuration to your `configuration.yaml` and restart Home Assistant:

Reviewer: Remote providers are for development and testing, they do not belong in ZHA docs/

Author: OK removed
